src/generator/operations.ts

```diff
@@ -808,22 +807,43 @@ function createProtocolErrHandler(op: Operation, imports: ImportManager): string
     }
     return unmarshaller;
   };
-  if (op.exceptions.length === 1) {
-    text += generateUnmarshaller(op.exceptions![0], '\t');
+  // fold multiple error responses with the same schema into a single unmarshaller.
+  const foldedMap = new Map<Schema, Array<string>>();
+  // create a dummy schema for schemaless errors
+  const genericErr = new Schema('generic', 'generic', SchemaType.Object);
+  for (const exception of values(op.exceptions)) {
+    let errSchema = genericErr;
+    if (!exception.language.go!.genericError) {
+      errSchema = (<SchemaResponse>exception).schema;
+    }
+    if (!foldedMap.has(errSchema)) {
+      foldedMap.set(errSchema, new Array<string>());
+    }
+    for (const statusCode of values(<Array<string>>exception.protocol.http!.statusCodes)) {
+      foldedMap.get(errSchema)?.push(statusCode);
```

Comment:
Why the `?` instead of `!`? Curious because of the flow I thought you could assume `!` this.

Reply:
Fixed (must have been an auto-complete thing).